inspector.py

```diff
@@ -82,7 +82,9 @@ def printAncestors(particle, ancestors=[], verbose=True):
         if abs(mum.pdgId())<8 or \
            abs(mum.pdgId())==21 or \
            abs(mum.pdgId()) in diquarks or\
-           abs(mum.pdgId()) in excitedBs: continue
+           abs(mum.pdgId()) in excitedBs or\
+           abs(mum.eta()) > 1000: # beam protons
```

Comment:
I think this would get in the way of cases where the first ancestor is, for example, a b baryon, that have PDG IDs > 5000.
I'd rather exclude protons explicitly, or see if some statusFlags could help telling apart beam protons from protons in decays

Reply:
Why is pdgId relevant? That is a cut on eta.
The statusFlag however is a good idea, I need to check whether the beam has a special status

Reply:
I don't think we can rely too much on the status, from my understanding of the documentation it can change based on the generator, and we use multiple ones

Reply:
> Why is pdgId relevant? That is a cut on eta.
sorry, I've (mis)read too quickly, I take it back

Reply:
Should I change anything else? Status does not help

Reply:
I guess this is ok, we can let Leonardo test it. The output will clearly tell if this is all or more is needed. (side note statusFlags != status, https://github.com/cms-sw/cmssw/blob/master/DataFormats/HepMCCandidate/interface/GenStatusFlags.h)